manifests/vcsscm/git.pp

```diff
@@ -45,6 +47,74 @@
           source => 'puppet:///modules/software/vcsscm/git/bash-git-prompt',
         }
       }
+
+      if $gitconfig or $gitignore {
+        file { ['/etc/skel', '/etc/skel/.config', '/etc/skel/.config/git']:
```

Comment:
This declaration introduces potential conflicts ("Duplicate declaration ... is already declared ... cannot redeclare") with modules that manage the same paths, e.g.

> Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Duplicate declaration: File[/etc/skel/.config] is already declared in file /etc/puppetlabs/code/environments/production/modules/nextcloud/manifests/client.pp:34; cannot redeclare at /etc/puppetlabs/code/environments/production/modules/software/manifests/vcsscm/git.pp:52 at /etc/puppetlabs/code/environments/production/modules/software/manifests/vcsscm/git.pp:52:9 on node puppet-testing.comsolit.local

The only feasible solution I can think of is running `mkdir -p` on the _base path_ (i.e. `mkdir -p /etc/skel/.config` in our case), and only manage the actual final target path with `file { ... ensure => directory }`.

Can you think of a better solution? For Window we clearly need to code this separately.